### PR TITLE
GAD1: added edit deapartment info and draft feature to announcement 

### PR DESCRIPTION
**Pull request by CHOUHAN-MOHIT**
- [x] added draft feature to announcement
- [x] fixed page not found error

### CODE REVIEW

**Comment by CHOUHAN-MOHIT** [comment]
@akshatnema

**Comment by akshatnema** [comment]
Require approval from reviewer.

**Comment by akshatnema** [comment]
Review required from @PranshuNayak and @samay-rgb

**Review by samay-rgb — APPROVED** [review]

**Review by PranshuNayak — APPROVED** [review]
LGTM!

**Comment by CHOUHAN-MOHIT** [comment]
@akshatnema added some more changes in department templates.

File: FusionIIIT/applications/department/views.py [hunk 1]
```diff
@@ -725,6 +755,44 @@ def deny(request):
...
+        department_ = request.POST.get('department')
+        is_draft = request.POST.get('is_draft')
+        ann_date = date.today()
+
+        # print("id:" , id)
+        # print("batch:" , batch)
+        # print("programme:" , programme)
+        # print("message:" , message)
+        # print("department_:" , department_)
+        # print("is_draft:" , is_draft)
```
Comment (akshatnema) [inline FusionIIIT/applications/department/views.py]:
```suggestion
```

File: FusionIIIT/applications/department/views.py [hunk 2]
```diff
@@ -725,6 +755,44 @@ def deny(request):
...
+        obj.batch=batch
+        obj.programme=programme
+        obj.message=message
+        obj.upload_announcement=upload_announcement
+        obj.department = department_
+        obj.is_draft = True if is_draft == "true" else False
+        obj.ann_date=ann_date
+
+        obj.save()
+        # department_notif(usrnm, recipients , message)
```
Comment (akshatnema) [inline FusionIIIT/applications/department/views.py]:
```suggestion
```

File: FusionIIIT/templates/department/edit-draft.html [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+        var formData = new FormData();
+        formData.append('id', id);
+        formData.append('message', allFields['announcement']);
+        formData.append('batch', allFields['batch']);
+        formData.append('programme', allFields['programme']);
+        formData.append('upload_announcement', upload_announcement);
+        formData.append('department', allFields['department']);
+        formData.append('is_draft', "true");
+        // console.log(allFields['batch']);
+        // console.log("success");
```
Comment (akshatnema) [inline FusionIIIT/templates/department/edit-draft.html]:
```suggestion
```

File: FusionIIIT/templates/department/edit-draft.html [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+            },
+            error: function (data, err) {
+                alert('Some error occured!!');
+
+            }
+        });
+    };
+
+    function draft_dft(event) {
+        // console.log(document.getElementById("btn-publish-dft").getAttribute("data-id"));
```
Comment (akshatnema) [inline FusionIIIT/templates/department/edit-draft.html]:
```suggestion
```

File: FusionIIIT/templates/department/edit-draft.html [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+        event.preventDefault();
+        var id = document.getElementById("btn-publish-dft-{{stu.id}}").getAttribute("data-id")
+        var $form = $('#draft-form-{{stu.id}}'),
+            allFields = $form.form('get values');
+        var upload_announcement = $('input[name="upload_announcement"]')[0].files[0];
+        // console.log(allFields);
```
Comment (akshatnema) [inline FusionIIIT/templates/department/edit-draft.html]:
```suggestion
```

File: FusionIIIT/templates/department/sm_dep.html [hunk 6]
```diff
@@ -108,15 +129,38 @@ <h3>
                                 </p>
                             </li>
                         </ol>
-                    </span>
+                    </span> {% endcomment %}
```
Comment (akshatnema) [inline FusionIIIT/templates/department/sm_dep.html]: If you are making this part of code as comment, kindly remove the code and don't make it as comment.

**Review by akshatnema — CHANGES_REQUESTED** [review]

**Review by akshatnema — APPROVED** [review]

**Pull request merged**